pkg/service/gcp/cluster.go

```diff
@@ -0,0 +1,259 @@
+package gcp
+
+import (
+	"context"
+	"fmt"
+
+	container_proto "google.golang.org/genproto/googleapis/container/v1"
+	"google.golang.org/grpc/codes"
+
+	"github.com/davecgh/go-spew/spew"
+	"github.com/googleapis/gax-go/v2/apierror"
+	"github.com/pkg/errors"
+	"gitlab.com/netbook-devs/spawner-service/pkg/service/constants"
+	"gitlab.com/netbook-devs/spawner-service/pkg/service/labels"
+	"gitlab.com/netbook-devs/spawner-service/pkg/service/system"
+	proto "gitlab.com/netbook-devs/spawner-service/proto/netbookai/spawner"
+)
+
+//getParent retrive getParent path for clusters
+func getParent(projectId, location string) string {
+	return fmt.Sprintf("projects/%s/locations/%s", projectId, location)
+}
+
+func getClusterFQName(projectId, region, cluster string) string {
+	return fmt.Sprintf("projects/%s/locations/%s/clusters/%s", projectId, region, cluster)
+}
+
+func (g *GCPController) createCluster(ctx context.Context, req *proto.ClusterRequest) (*proto.ClusterResponse, error) {
+
+	cred, err := getCredentials(ctx, req.AccountName)
+	if err != nil {
+		return nil, errors.Wrap(err, "createCluster ")
+	}
+
+	client, err := getClusterManagerClient(ctx, cred)
+	if err != nil {
+		return nil, errors.Wrap(err, "createCluster")
+	}
+	defer client.Close()
+
+	//cluster tags
+	tags := make(map[string]string)
+	for k, v := range labels.DefaultTags() {
+		tags[k] = *v
+	}
+
+	for k, v := range req.Labels {
+		tags[k] = v
+
+	}
+
+	np, err := getNodePool(req.Node)
+
+	if err != nil {
+		g.logger.Error(ctx, "failed to get the nodepool config from the spec", "error", err)
+		return nil, errors.Wrap(err, "CreateCluster: getNodePool returned error")
+	}
+
+	cluster := &container_proto.Cluster{
+		Name:           req.ClusterName,
+		Description:    "Spawner managed cluster",
```

Comment:
Can we add Netbook as well to the Description?

Reply:
we can add, but as a spawner is open source tool and we are targetting the community there, might not make sense to say netbook in standalone context, wdyt ?

Reply:
Netbook will just be a namespace, spawner is by netbook org so should not be an issue. Will be easier for users using Netbook connect to identify its run by netbook.

Reply:
that make sense too. 
so how would you like to word it? `netbook-spawner managed cluster`

Reply:
works

Reply:
updated.